### ISSUE TITLE
LiveTV Refresh Guide scheduled task completely wipe out any scheduled recordings

### ISSUE THREAD

**Issue by osscombat** [jellyfin/jellyfin#12979]
### This issue respects the following points:

- [x] This is a **bug**, not a question or a configuration issue; Please visit our forum or chat rooms first to troubleshoot with volunteers, before creating a report. The links can be found [here](https://jellyfin.org/contact/).
- [x] This issue is **not** already reported on [GitHub](https://github.com/jellyfin/jellyfin/issues?q=is%3Aopen+is%3Aissue) _(I've searched it)_.
- [x] I'm using an up to date version of Jellyfin Server stable, unstable or master; We generally do not support previous older versions. If possible, please update to the latest version before opening an issue.
- [x] I agree to follow Jellyfin's [Code of Conduct](https://jellyfin.org/docs/general/community-standards.html#code-of-conduct).
- [x] This report addresses only a single issue; If you encounter multiple issues, kindly create separate reports for each one.

### Description of the bug

I use an M3U tuner with XMLTV guide provider. The guide updates quite often, so I have a scheduled task for guide update every 1 hour.

The issue is that running this task completely reset any scheduled recordings, they just disappear from the recording schedule, making the whole DVR feature useless.

Nothing relevant in the logs to attach, the guide is updated fine with: 
`Emby.Server.Implementations.ScheduledTasks.TaskManager: "Refresh Guide" Completed after 2 minute(s) and 19 seconds`

### Reproduction steps

1. Setup an M3U tuner
2. Setup an XMLTV guide source for the tuner channels
3. Schedule any future shows from the Guide to record
4. Run LiveTV Refresh Guide task
5. Notice all sheduled recordings vanish

### What is the current _bug_ behavior?

All sheduled recordings vanish with TV guide refresh

### What is the expected _correct_ behavior?

No scheduled recordings vanish with TV guide refresh

### Jellyfin Server version

10.11.1+

### Specify the build version

10.10.7

### Environment

- OS: Linux
- Docker CE with official Jellyfin Hub image:

### Jellyfin logs

n/a

**Comment by diogotcorreia** [comment]
I've changed lists meanwhile and this has stopped being a problem for me. Perhaps the new list includes an ID of some sort for guide entries?

@osscombat is this still an issue for you?

**Comment by osscombat** [comment]
> I've changed lists meanwhile and this has stopped being a problem for me. Perhaps the new list includes an ID of some sort for guide entries?
> 
> [@osscombat](https://github.com/osscombat) is this still an issue for you?

Still an issue for the latest version, unfortunately. I have a pretty simple EPG without any IDs, never seen any reference that EPG XML should contain any:

<?xml version='1.0' encoding='utf-8'?>
<tv generator_info_name="Local">
<channel id="Channel1">
<display-name>Channel1</display-name>
</channel>
<programme start="20250308053700 +0000" stop="20250308055800 +0000" channel="Channel1">
<title lang="en">News</title>
<icon src="https://imgurl" />
<desc>Description</desc>
</programme>

**Comment by diogotcorreia** [comment]
@osscombat You're right, there are no IDs in my EPGs either. I've tried out my old EPG source and it also works now for some reason 🤔 
Sadly can't reproduce this anymore :/

**Comment by joe-jf** [comment]
I have this exact issue.  I also noticed if I set any channels as "favorites" they will be cleared with a guide refresh.  I also have a simplistic XMLTV from my provider. 

Does anyone know of a fix?

**Comment by osscombat** [comment]
Did some experimenting with XMLTV to exclude the images pre-cache errors. The guide artwork is being downloaded without single error, but it still looks like the Refresh Guide task just delete all the channels. The simplified log for scheduling the record and then Refreshing:

```
[2025-05-27 15:27:38.634 +03:00] [INF] [48] Jellyfin.LiveTv.Timers.TimerManager: Creating recording timer for "563b647e748745b18614f9a0d27be06d", "#############" on channel "m3u_b0fdccb5444e0f8914fb238c3299f2aed2129026294e22201bb876c9d4d9b360". Timer will fire in "2907.3560848266666" minutes at 05/29/2025 13:00:00
[2025-05-27 15:35:19.730 +03:00] [INF] [20] Jellyfin.LiveTv.TunerHosts.M3UTunerHost: Parsed channel: "Bloomberg"
[2025-05-27 15:35:19.734 +03:00] [INF] [66] Jellyfin.LiveTv.Listings.XmlTvListingsProvider: xmltv path: "http://iptv.lan/epg.xml.gz"
[2025-05-27 15:35:19.752 +03:00] [INF] [66] Jellyfin.LiveTv.Listings.ListingsManager: Found epg channel in "XmlTV" null "Bloomberg" "tvg-id-17"
[2025-05-27 15:35:19.804 +03:00] [INF] [66] Jellyfin.LiveTv.Guide.GuideManager: Refreshing guide with 6 days of guide data
[2025-05-27 15:35:19.805 +03:00] [INF] [66] Jellyfin.LiveTv.Listings.XmlTvListingsProvider: xmltv path: "http://iptv.lan/epg.xml.gz"
[2025-05-27 15:35:34.620 +03:00] [INF] [73] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "LiveTvChannel", Name: "Bloomberg", Path: "", Id: 1d9e1339-6229-2ab8-9dc1-4dbff119b298
[2025-05-27 15:35:40.907 +03:00] [INF] [73] Emby.Server.Implementations.ScheduledTasks.TaskManager: "Refresh Guide" Completed after 0 minute(s) and 24 seconds
```

**Comment by osscombat** [comment]
The same XMLTV source works and schedules fine in NextPVR instance, so this is purely a Jellyfin bug.

**Comment by osscombat** [comment]
Not fixed

пт, 3 окт. 2025 г., 05:04 jellyfin-bot ***@***.***>:

**Comment by osscombat** [comment]
bump for version 10.11.1

**Comment by Ruchamimi** [comment]
I have similar issue. I use threadfin as M3U proxy for filtering channels and EPG data, as well as buffering streams for better stability in JF. I don't know why it wipes all scheduled recordings at each live TV guide refresh, as it seems the channel list and IDs in the m3u seems to be stable from one day to another.
I will try to replace threadfin with dispatcharr or TvHeadend to check if it brings stability, if anyone has experience with this tools as proxy I would be interested.

EDIT : Using dispatchaar seems to solve the issue, the recordings remains after 1 daily live TV guide refresh. It looks like the m3u and/or the EPG it provides is built with enough stability for Jellyfin. I will report if I loose recording schedule after a few days.
